fennel/test_lib/query_engine.py

```diff
@@ -0,0 +1,328 @@
+from collections import defaultdict
+from datetime import datetime
+from functools import partial
+from typing import Dict, List, Union, Optional, Any
+
+import numpy as np
+import pandas as pd
+from frozendict import frozendict
+
+import fennel.datasets.datasets
+from fennel.featuresets import Extractor, Feature, Featureset, is_valid_feature
+from fennel.gen.featureset_pb2 import (
+    ExtractorType as ProtoExtractorType,
+)
+from fennel.gen.schema_pb2 import Field, DSSchema, Schema
+from fennel.lib.schema import data_schema_check, get_datatype
+from fennel.test_lib.branch import Entities
+from fennel.test_lib.data_engine import DataEngine
+from fennel.test_lib.test_utils import cast_col_to_dtype, parse_datetime
+
+
+class QueryEngine:
+    def __init__(self): ...
+
+    def lookup(
```

Comment:
Add comments to all the functions please. For eg whats the difference b/w this lookup and lookup in data engine. I wonder if it can be named differently?  like run_lookup, cause lookup is actually provided by DataEngine?

Reply:
will add in all places.